src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -152,6 +156,34 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
     this.timezoneService.getTzVersion(); // import timezone service to load timezone data
   }
 
+  handleAutoSave(event: { id: string, model: QuestionSubmissionFormModel }): void {
+    clearTimeout(this.autoSaveTimeout);
+    this.autoSaveTimeout = setTimeout(() => {
+      const savedData = JSON.parse(localStorage.getItem('autosave') || '{}');
+      const clonedModel = {
+        ...event.model,
+        hasResponseChangedForRecipients: Array.from(event.model.hasResponseChangedForRecipients.entries()),
+        isTabExpandedForRecipients: Array.from(event.model.isTabExpandedForRecipients.entries()),
+      };
+      savedData[event.id] = clonedModel;
+      localStorage.setItem('autosave', JSON.stringify(savedData));
+    }, this.autoSaveDelay);
+  }
+
+  loadAutoSavedData(questionId: string): void {
+    const savedData = JSON.parse(localStorage.getItem('autosave') || '{}');
```

Comment:
I think localStorage related methods (parsing for the item, updating the item and deleting the item) can be extracted into their own function(s). Would make it easier to maintain as these are used quite a few times here, for e.g. if we want to change the key from "autosave" to something else.